### PR TITLE
Introduce service stable logic to ensure ECS workload available

### LINKED ISSUE #4478: ECS deployment should wait until new version is rolled out completely
**What would you like to be added**:

**Why is this needed**:

Currently, PipeCD exits immediately after preparing/updating new TaskDefinition and ServiceDefinition.
Because of that something users have to check the actual rolling out status on AWS console.

Therefore, I propose adding a new flag in ECS deployment configuration to force checking the rolling out status before mark the deployment as completed.

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #4478 

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

**Comment by khanhtc1202** [comment]
@kentakozuka plz

File: pkg/app/piped/platformprovider/ecs/client.go [hunk 1]
```diff
@@ -29,9 +30,17 @@ import (
...
 	"github.com/pipe-cd/pipecd/pkg/app/piped/platformprovider"
+	"github.com/pipe-cd/pipecd/pkg/backoff"
 	appconfig "github.com/pipe-cd/pipecd/pkg/config"
 )
 
+const (
+	// Retry wait service stable check.
+	retryServiceStable = 40
+	// Interval wait service stable check.
+	retryServiceStableInterval = 15 * time.Second
```
Comment (kentakozuka) [inline pkg/app/piped/platformprovider/ecs/client.go:41]: Can you tell me why this timeout is 10mins(40 * 15sec)?
The default deployment timeout is now 6 hours.

**Review by kentakozuka — COMMENTED** [review]
Nice :)
Left a comment.

File: pkg/app/piped/platformprovider/ecs/client.go (hunk 1 above)
Comment (khanhtc1202) [reply pkg/app/piped/platformprovider/ecs/client.go:41]: This follows the AWS cli implementation, where they use the above numbers. Of course, we can wait up to 6 hours, but I just feel like it's overdoing.
https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ecs/wait/services-stable.html#description

Comment (kentakozuka) [reply pkg/app/piped/platformprovider/ecs/client.go:41]: OK 👍 Thanks

**Review by kentakozuka — APPROVED** [review]
Super!
Let's go 🚀

**Pull request merged**